### ISSUE TITLE
Transhumanism rework proposal, part 1: Bionic slots

### ISSUE THREAD

**Issue by I-am-Erk** [CleverRaven/Cataclysm-DDA#28273]
**Is your feature request related to a problem? Please describe.**  
Bionics and mutations represent an overall category of transhumanism, which should start in the mid-game and progress into the late-game, allowing the player to slowly develop into a superhuman or a monster. Currently the transhumanism progression suffers from a number of problems. Key among these in my opinion:
1. It tends to be all-or-nothing, especially with bionics. There is little decision making to the installation of CBMs, aside from initially having to ration autodoc access and anaesthetic. As access to CBMs becomes less of the gateway (through things like building your own clinic), the gating system needs to ask for player *choice*, which is more meaningful and fun than lucky finds anyway.
2. Mutations offer a fun gambling subgame, but one where the risk-rewards are very poorly balanced currently. The gambling aspect of mutations needs to be preserved, but there needs to be a higher element of player control, again for the purpose of proper decision making. As well, there should be more of a realistic feel, as opposed to the current insta-mutation.
3. Mutations and bionics currently are fully complementary, allowing you to get the best of both worlds with little to no tradeoff.
4. As Kevin points out, "There is no meaningful tradeoff between acquiring mutations and CBMs vs refraining from doing so. If you can acquire them, it's a given that you should do so, because it's only plusses. Some of the mutations have drawbacks, some are significant, but it's (IMO correctly perceived) as something to manage by trying to get the right mutations, not avoiding them entirely."

**Describe the solution you'd like**  
This issue will address primarily bionics, see https://github.com/CleverRaven/Cataclysm-DDA/issues/28277 for mutations - although I think the two problems need to be considered at once.

Bionic slots were previously introduced as a way to limit bionics, but the slot system has been largely abandoned with little excitement about reintroducing it. I propose we modify the system with the following characteristics.
- Expand the granularity of bionic slots in the torso and head, making the following slots: Arms, legs, head, eyes, chest, guts.
- Each slot site gets two slot numbers, "safe" and "max". For example you might have a "safe slot limit" of 15 on your eyes, and a "max slot limit" of 30.
- Mutations are flagged to which body parts they change (they already do this to some degree). Mutations on a body part decrease the max slot limit of that body part. The safe slot limit shouldn't change except in extraordinary circumstances (ie. some very extreme mutations might drop it, but that would be a special feature of that mutation), because the idea is that the player should be able to stay under that limit and not have to worry about running into system problems.

Numbers of slots:
This is something big and outside the scope of this PR because it needs careful examining of all the CBMs and where they'd fit.
- Some CBMs might be small enough to not require any slots, or to have a negligible slot cost - clock, for example, or joint tension ratchet, could both easily be a single slot or less. In the end though, you should be able to stay within Safe Slot Limit and have a cool selection of different bionics... just nowhere near a tool for every occasion. 
- Some very powerful bionics might exceed the safe slot limit on their own, but those should be very much the rare exception.

Installation and slots:
- A body part with no bionics in it at all gives a boost to surgeon/autodoc success rates.
- After the first bionic, and until the safe limits, success rates are baseline
- After the safe limit, difficulty of installation steadily increases

Costs of too many bionics:
- Increasing your overall number of bionics to a high value, regardless of the number of body parts, causes a permanent effect. This should worsen as you get more bionics.

*Open Skin*: This is the only effect you can possibly get if your bionics are all within the safe limit, and it should never be too serious in that case. Causes an increased risk of infection that increases the more bionics you get. Increases fairly quickly when you pass the safe bionic limit.

*Nerve Pain*: starts to increase as you pass the safe bionic limit in any limb. Causes at first a small amount of pain, gradually increasing. Can be negated with increasingly high doses of anticonvulsant meds, which would need to be added to the game (gabapentin eg.)

*Specific*: Certain CBMs might have specific downsides that would happen if they were installed in unsafe slots. For example, cybernetic armour might reduce strength if you're at 50% or more of your max slot limit when you install it, due to having to remove muscle tissue to make room for it. Once a CBM gets this downside, it's part of the CBM, even if you reduce your slot burden later (unless you fully remove the problem CBM, which would presumably involve muscle grafts to replace the missing tissue). An autodoc probably wouldn't warn you about these costs, but a human surgeon absolutely should.

*Other*: Slot-specific problems that increase after the max limit is hit should be added too, for example "loss of taste" for the guts slot, reducing the enjoyment of food progressively as the amount increases; or a "clicking legs" for the legs slot, causing increased noise with movement.

Removal:
CBM removal needs to be part of the slot system. I suggest this is the ideal place for the poor ol' Mr Stem Cell to return, along with an autodoc or surgeon to do the rest of the job.

Full-Conversion:
this is the real special element that makes it all fit together. This needs to be a choice in the late game, where specialized factions and equipment can be used to change your entire body into a robot, giving you infinite slots. You have to purify yourself back to baseline for this to work, and mutations will no longer be possible for you. You get a series of traits related to being a robot now (eg. almost no need for food, no longer at risk of infection because your brain is in a jar, but also no baseline healing, need constant supply of scrap metal to run your nano bots to repair you, and risk of dying if you run out of power).

****Summary/Final goal:****
if you want a bit of cybernetics, use CBMs up to your safe slot limit. These should be available in the early mid game, relatively safe to install with help, and accessible. Mutants can use them too without much concern. You will at most face a little bit of increased infection risk, and only if you really push the edges of safe.

If you want a lot of cybernetics, use CBMs going towards your max slot limit. You will get some penalties, but they can nearly all be negated with drugs, and the ones that can't will be far from gamebreaking (eg losing sense of taste, or making a noise when you move). Your max slot limit should be quite high, allowing late-midgame cyborgs to be similar to current mid game characters. No, you can't have all the cybernetics, and you definitely can't have unlimited powah, but you should be able to feel like a real cybernetic monster. The primary tradeoff is that as you get higher and higher in your cyber-limit, you are more and more limited in what mutations might still be compatible with your altered body.

In the end, even a full conversion cyborg shouldn't be as raw-powerful as a current cyborg is 2 years of game time in. You won't be able to stop time for a month of game time because you have 100k battery storage. However, adding full conversion to replace this power offers far more *fun* choices rather than powerful. Stuff like switching bodies to one more suited to a purpose; having spider legs that let you climb on walls; having a body that is basically a tankbot; having three different awesome gun arms that can retract into your crazy robot torso.

Upcoming: I will do a similar post about mutations in my next break, which will be complementary to this one in some ways.

**Comment by I-am-Erk** [comment]
https://docs.google.com/spreadsheets/d/1po7bBB6VR3tpJKcG7kmlTnGJs7GYiKWXpFGH-YKx8Ac/edit?usp=sharing is a relevant google doc put together by Brian, which offers a lot of good information on future balancing of CBMs. It's not directly related to what I'm talking about now, but needs to be a consideration in slot costs and balancing.

**Comment by kevingranade** [comment]
I'd add one more key problem, there is no meaningful tradeoff between acquiring mutations and CBMs vs refraining from doing so. If you can acquire them, it's a given that you should do so, because it's only plusses.  Some of the mutations have drawbacks, some are significant, but it's (IMO correctly perceived) as something to manage by trying to get the right mutations, not avoiding them entirely.

**Comment by I-am-Erk** [comment]
That is true. And in my system I do assume that in general there is an amount of CBMs you can have with no tradeoff. I think that's OK, I assume these wouldn't reach market if they couldn't be installed safely in small numbers.

**Comment by Fris0uman** [comment]
Could the slots have layers ? 
>  Arms, legs, head, eyes, chest, guts.

That's almost the slots of the clothing system, in a comparable way we could have layer that goes : Skin/Blood  vaissel/Peripheral Nervous System/Bones on the arms and legs and
Skin/ Blood Vaissel/ Peripheral Nervous system/Bones /Organs for torso and head. And then you have limited amount of space or encumbrance or wathever it's going to be called for each layer. Or is it too much detail ?

> there is no meaningful tradeoff between acquiring mutations and CBMs vs refraining from doing so.

Should there be a tradeoff because it's good game design or because it would make sense realistically ? It's a real question, it's not obvious to me why being post-human in the cataclysm could be bad aside from normal human not liking you. 
Unless : 
>I assume these wouldn't reach market if they couldn't be installed safely in small numbers.

What I understand of the lore is the world pre-cataclyms is a bit on the careless side and don't value much safety and human life so the lore might point to CBMs no being very safe.

>Cost of too many bionics

This seem pretty reasonnable, but I'm not too sure about `open skin` if the things are intalled and sealed properly it should be ok. In the real world laboratory animals live for months with electrodes in their brain without infections (If the surgery were correctly done...) so it's not obvious that having CBMs just at capacity should leave you open to infections. But on the other hand "being just at capacity" might mean having already quite a lot of foreign bodies in you.

>An autodoc probably wouldn't warn you about these costs, but a human surgeon absolutely should.

But when using the autodoc you are the human surgeon and the autodoc is your high grade assistant so surely if you were going to remove significant portion of the patient muscle it would tell you ? Or at least with enough skill you would understand yourself what you're going to do ?

>risk of dying if you run out of power

Being dependant on power could be a drawback that happens before you become a full cyborg, for example if you get bionic legs or arms they just stop functionning and you're crippled.

Otherwise your ideas seem pretty good 👍

**Comment by Brian-Otten** [comment]
Here's the spreadsheet i made after testing every bionic in the game, i tested them for current functionality and power usage, estimated their realism levels and what they should be doing if made more realistic, compared game balance, and left notes on what changes seemed wise to me.
https://docs.google.com/spreadsheets/d/1po7bBB6VR3tpJKcG7kmlTnGJs7GYiKWXpFGH-YKx8Ac/edit?usp=sharing

**Comment by I-am-Erk** [comment]
> Could the slots have layers ?

It's a cool idea, but I think avoiding too much detail makes this easier to balance, by handwaving away "how does all this fit in my ____" a little more. Plus, it becomes harder to decide exactly which slot something should fill if there are too many options.

> Should there be a tradeoff because it's good game design or because it would make sense realistically ? It's a real question, it's not obvious to me why being post-human in the cataclysm could be bad aside from normal human not liking you.

Mostly good game design. Trade-off is one of the most important aspects of designing an interesting game system. Each decision should come at a price. For balance reasons, trade-off helps to limit our current geometric progression problem.

> > I assume these wouldn't reach market if they couldn't be installed safely in small numbers.
> What I understand of the lore is the world pre-cataclyms is a bit on the careless side and don't value much safety and human life so the lore might point to CBMs no being very safe.

That makes sense for xedra and the government, but CBMs are also available at the consumer end, and people don't tend to pay big bucks for things that leave them crippled and in pain. Not for long, anyway.


> This seem pretty reasonnable, but I'm not too sure about `open skin` if the things are intalled and sealed properly it should be ok. In the real world laboratory animals live for months with electrodes in their brain without infections (If the surgery were correctly done...) so it's not obvious that having CBMs just at capacity should leave you open to infections. But on the other hand "being just at capacity" might mean having already quite a lot of foreign bodies in you.

Lab rats don't go wading through gore and zombies without access to a clean shower. The "Open skin" option doesn't appear until you have so many CBMs that things are starting to get crowded.

Also, some - I think a majority - of CBMs require skin access. Finger lasers, integrated toolsets, even flashlights have to poke out of the skin to do their job. I assume that part of achieving CBM technology was figuring out how to deal with biofilms reliably, but nevertheless anything indwelling in the skin will always carry an infection risk.

**Comment by Brian-Otten** [comment]
Also, quick thought on pure cyborg mode, I think it would probably be coolest if even mutants could go brain in jar mode, just make it forcibly take away all physical mutations, but keep the mental mutations. This leaves roleplay opportunities such as being an "apex predator" robocop, etc.
It also means no stacking the strongest physical mutations with cbms, which was the main balancing worry.

**Comment by Fris0uman** [comment]
>but nevertheless anything indwelling in the skin will always carry an infection risk.

I was going to try to discuss that, and then I though of doing a minimum of research and first result show that you're right  : [cochlear implant risks](https://www.fda.gov/MedicalDevices/ProductsandMedicalProcedures/ImplantsandProsthetics/CochlearImplants/ucm062843.htm#b)

**Comment by kevingranade** [comment]
> Full-Conversion

I think there's a nice intermediate version of this we could support, which would most likely swap out the player's digestive system and lungs (because of bulk) in return for making available a large number of torso slots.
This would be a VERY invasive procedure, and might require something like a faction surgical team or a small team operating an autodoc, and would come with a number of permanent penalties, such as higher chance of infection, ongoing power cost from operating organ replacement modules, and restricting diet to certain things the artificial digestion system can handle.  It would also be incompatible with a number of mutations.  On the other hand, it would open up a significant number of slots, and would likely be a prerequisite for certain extra powerful CBMs.

For full cyborg, I'm thinking the operating requirements would be even more extreme, with only final stages of certain factions capable of offering it at all.

> Should there be a tradeoff because it's good game design or because it would make sense realistically?

Yes to both, from a game design point of view, it's bad to restrict which types of characters are capable of being "the best", and from a realism point of view, it's unlikely in the extreme that you can reconfigure a living creature and end up significantly better in every way with no tradeoffs.

> I think it would probably be coolest if even mutants could go brain in jar mode, just make it forcibly take away all physical mutations

I'm not so sure about leaving mental mutations, but I'm not going to rule it out.  I do think that instead of purifying back to your base state it would be simpler for the procedure to cancel all the mutations.

**Comment by Fris0uman** [comment]
> It would also be incompatible with a number of mutations.

So mutations restrict the CBMs you can install but what happens when you go the other way ? What happens if you're chokefull of CBMs and start mutating ? Do your CBMS fall out ? Do they just start malfunctioning ?

**Comment by kevingranade** [comment]
I said incompatible, but I'm not sure how that would work out in practice.  It might require you to purify it away first, or it might forcibly cancel it.
As you point out, we do need a mechanism for what happens when you take mutagen and it picks a mutation you can't have anymore.  Unless something better comes along, I'm assuming that presence of a CBM is going to simply prevent acquisition of mutations it conflicts with.

**Comment by I-am-Erk** [comment]
If you're loaded up with CBMs and mutate that body part so extremely that they don't fit anymore, you'd probably destroy and eject your CBMs, damaging the limb.

Luckily given my proposed mutation system, you will have warning. If your arms is loaded with CBMs, and you drink mutagen and your arms start to itch, you might want to take purifier.

**Comment by Photoloss** [comment]
> from a realism point of view, it's unlikely in the extreme that you can reconfigure a living creature and end up significantly better in every way with no tradeoffs.

There is a difference between "every way" and "any way" though. Only grabbing one or two CBMs, just like lucking out on one or two benign mutations, would not always "realistically" come with a downside. The surgical procedure itself might cause some side effects but subtituting your appendix for a vitals monitor does not actually impede your body. 

In the same vein there is no benefit to *not* using either martial arts or dedicated melee weapons. Adding a penalty simply makes no sense, learning to fight or equipping yourself is better than doing neither. Plus once NPCs actually do things the general social detriments will already be a tradeoff as opposed to "yay I get to growl at my meatshield before I eat it!"

**Comment by Fris0uman** [comment]
>Only grabbing one or two CBMs, just like lucking out on one or two benign mutations, would not always "realistically" come with a downside.

Good thing that this is exactly what's proposed here then.

**Comment by MichaelEricOberlin** [comment]
Just a thought as a real-life physicist who has worked with real-life neuroprosthetics (effectively bionics). You aren't kidding about "open skin" being a problem; as it turns out, a man can lose his life to things like that, even if the incision is small. Additionally, the human body is weirdly good at rejecting anything that's _just not supposed to be there_, healing over it; so, we actually don't do open incisions at all—real-world bionic arms and legs communicate to a second piece, beneath the skin, with magnetic pulses, which generally works with the nerves. (I avoid saying "electromagnetic" because it implies more current than is actually used.) They're two-part now, one external, and one internal; and we have quite a few people using things like this very effectively. (Think of an EEG on your limb stump, and you've got the idea.)

The only part that would get science-fictiony is the accelerated mapping of nerve impulses in the stump, to limb movements; that generally takes about a year of regular calibration right now. (Which is part of why you aren't seeing them everywhere, or at least everywhere somebody's lost a limb, yet.) Personally i think that's one of those one-line-of-dialogue corrections though, particularly given the time frame of the game. Super-advanced pattern-recognition algorithms or something like that.

There was an article written about ten years ago, in Scientific American, maybe 2007, that did a pretty good job covering all of the details; it seems to be almost impossible to find now. I did find [this](https://www.the-scientist.com/features/neuroprosthetics-36510) though, which is much more recent but seems to send the message home. Honestly I've never even heard of that publication, but the article seems to do the job.

Having actual incisions on the limbs at that point in history would just be kind of backward for me.

**Comment by I-am-Erk** [comment]
in real life we definitely don't allow open incisions at all. I actually didn't know about cool magnetic pulse communication stuff (dammit jim, I'm a doctor not a physicist), so that can help explain why you'd be able to get something like an integrated toolset installed without developing Open Skin.

Regardless of the exact mechanics, it's basically inevitable that more bionics will increase infection risk. This is an issue we face in medicine all the time with existing inorganic additions like knee replacements already. The name of Open Skin could change to indicate this, or the description could change to indicate the intent, which is that it's a problem resulting from having too many bionic systems, not necessarily the result of any one system (things like skin breakdown over prosthetic sites as they are forced too near the surface would be a legitimate issue).

**Comment by kevingranade** [comment]
Alternately, maybe these bionics that cross the skin barrier shouldn't exist at all.
I'm looking at you integrated toolkit.

**Comment by KorGgenT** [comment]
i think i'm going to augment this discussion with a spreadsheet using the data in the above google doc. Maybe if i write out a "priority" (how much a player wants this function) "suggested slot limb" and a "power level or slot number description" we can hash out the actual balance of the CBMs themselves. I'll probably need a day or two to write it out, and as I haven't used every single CBM in the game once I get through it I could use some input.

**Comment by mlangsdorf** [comment]
Before we balance individual CBMs, I suggest we categorize the existing bionic professions on some kind of rough estimate: definitely safe, probably safe, possibly not safe, certainly not safe.  That kind of rough consensus up front might save some heartache on the back end.

**Comment by Photoloss** [comment]
> Here's the spreadsheet i made after testing every bionic in the game, i tested them for current functionality and power usage, estimated their realism levels and what they should be doing if made more realistic, compared game balance, and left notes on what changes seemed wise to me.

That spreadsheet contains some inaccuracies, are we as random participants allowed to correct it or should that be reserved for authors/admins?

- Blood Analysis has several hardcoded returns [here](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/src/bionics.cpp#L249) and should be able to detect most infections and parasites *long* before you yourself can. 
- Integrated Dosimeter has a unique automated/advance warning [here](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/src/player.cpp#L5863) and if it does not do so already should provide the full functionality of the radiation biomonitor item. Whether that is enough to warrant keeping it in the game is up for debate of course.
- There is a [recipe](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/data/json/recipes/recipe_ammo.json#L27) for crafting plutonium slurry from cells. I have not checked whether this provides any benefit and have no idea whether it *should* (greater surface area for faster processing?) but the claim in the spreadsheet is factually wrong.

Gameplay aside it seems really odd to me to remove the few bionics that are actually functional IRL as prototype medical diagnostics. Most of this is because the @ screen cheats a bit with what the player *should* be able to learn about themselves, especially at low first aid skill.

**Comment by RedPine91** [comment]
Adding to the discussion of bionic slots:  Going "Full Bionic" should be decided on a per-limb basis.

Full Bionic Body: Cyborg, Teen Titans.  Nearly all flesh has been completely replaced with machine.  As many bionics as desired can be installed, limited only by the bulk, weight, and power consumption desired. Mutations are impossible, as there is no flesh left to mutate in the first place.  

Full Bionic Limb: Edward Elric, Full Metal Alchemist.  One arm and one leg are full bionics, they can fit as many gadgets as can physically fit inside that limb.  The remaining limbs and torso are full human, and susceptible to mutations.  The bionic limbs cannot be mutated because there is no flesh to mutate in the first place.

Partial Bionic Partial Mutant: Wolverine, X-Men.  Cannot go full bionic, as existing mutations reject excessive CBMs.  However, as long as a bionic is within the "safe limit" - such as claws - it is compatible with the body.

I would propose the following implementation:

Safe Bionic Install:  Install a CBM within the safe slot limit.  
Unsafe Bionic Install:  Install a CBM up the max limit.  High risk, low cost.
Replace Limb:  Replace the limb with an entirely bionic one, in order to increase max slot limit. Low risk, high cost.

To go full bionic, you would need to replace ALL limbs, including the torso and head.

To replace a missing limb that has been replaced, you would need to acquire extreme or rare mutations that enable limb regeneration.  This would destroy any bionics.

**Comment by OldFriendEr** [comment]
So i had thoughts about the whole mutation vs bionics thing in general, to ground the practical changes to some sort of real reason. It would make sense to make a bionic tier list, depending on how complicated and how much space it takes in the body. 
My reasoning is that the bionics are carefully crafted for humans and no human-trees or human-spiders. Not only the bionic part must be compatible with the host body but the installation process too. I doubt many autodocs were made for mutants, maybe surgeons could help there.

Anyway to get to my point i propose to make a bionic tier list, so simple bionics like batteries can be installed to piratically anything, but things that depend on human anatomy like reflexes or alloy plating become incompatible, because it simply does not fit the intended purpose.

As for addressing the existing bionics I suggest damaging one of the installed high tier bionic to a faulty bionic, so it would force superhuman players feel bad about getting a random mutation by external causes, besides the RP reasons. (we could even do this last point without the transhuman rework)

**Comment by actual-nh** [comment]
> Anyway to get to my point i propose to make a bionic tier list, so simple bionics like batteries can be installed to practically anything, but things that depend on human anatomy like reflexes or alloy plating become incompatible, because it simply does not fit the intended purpose.

I'd say that enhanced reflexes would work... on parts that were still humanoid (not, say, tentacle arms), and not cumulatively with positive mutation effects. More generally, Medical and maybe Alpha mutations may be exceptions on whether bionics and mutations are compatible.

**Comment by PaulBlay** [comment]
There doesn't seem to be much on reworking the power CBMs here. A few things, to get a decent amount of power (which most bionics needs) you either have to have a lot of power CBMs or fewer Mark II Power CBMs. One problem is that this means taking up many slots, another is that drop rates for power CBMs should arguably be much higher. If (pretty much) everybody using CBMs needs a lot of them there should be more found ready to install, and more found when dissecting corpses.

I would suggest higher capacity for power CBMs, possibly heavier power CBMs and probably more of them to be found.

**Comment by anothersimulacrum** [comment]
> One problem is that this means taking up many slots

Power CBMs do not take up slots. They should, but they do not.

**Comment by andrewhr** [comment]
**TL;DR: limb replacement with ablative pockets and gunmod-like system as flavorful alternative to proposed slot system**

I've been thinking about this for a while, and Full Bionic Limb (as mentioned above) is the best in terms of gameplay, lore, and plausibility - with a twist to side-step the slot system proposed in this issue. Sorry in advance about the wall of text.

**The suggestion**

First, double down on bionics as organ and limb replacements. At first, we can have limbs (arms/hands and legs/feet); This limb replacement can have some baseline characteristics as far as the engine can support - different interactions with healing and muscle gain (as there is no muscle here), tunned limb scores, etc.

Then the twist: think of those bionic limbs as "platforms". I imagine something like the gun mod and pocket systems. Some examples:
* monoblade is a forearm attachment
* integrated tools is a hand or finger attachment
* bionic armor work like ballistic vests plates

Most CBMs would then migrate to this system of bionic attachments. Even batteries mean we have a "natural" upper limit of how many you can have. 

The player could buy or upgrade to better limbs (better baseline, more slots, support for better attachments). The attachments can be crafted from existing tools and adapted to work in the bionic attachment (give some manual sold by Exodii).

**What I like about it**

**Gameplay**: replacing libs and organs already works as blockers to certain mutation traits. As the proposed system requires partial replacements, we have the transhuman trade-off by definition - do little of it to tap both power sources but doing too many locks you out.

It also has room to lean more on crafting (by doing some attachments), which the new mutation system does, and it is a good design choice. Of course, the replacement themselves and more advanced attachments should require quests and a good relationship with Exodii.

**Lore-wise**: Exodii are full cyborgs, so a complete metal replacement of your body, while very invasive, is fit for how they operate. Doing so increasingly locks mutations, which resonates with how they protected themselves against the Blob effects. The scavenging of tools to craft attachments also fits their flavor of scavengers and adapting to new worlds and aligns with how we acquire their faction-specific gear (buy recipe, then DIY). 

**Plausibility**: here are two threads on Reddit about current CBMs and how they don't make much sense. The limb (and organ) replacement achieves a good balance of suspending disbelief to allow current CMB capabilities while not violating those points too much (which I agree with).

**Further thoughts**

The examples are for limbs, but everything should work like this. Some brain dump on how this could work:
* "Install" of replacement works as current CBMs should (anesthetics, autodocs, maybe instructions).
* The replacement itself is non-craftable, acquired by quests and/or dismembering zomborgs and dead cyborgs. Some quests may be all about that recovery process.
* Modding a limb could be done by yourself, given tools and skills. At the end is maintenance work, necessary for weapons and armor as they get damaged from use.
* Modding organs - internal ones, or things like eyes - require cirurgical means, and should work like current CBM system, with the extra requirement of requiring base organ replacement.
* Most simple attachments and armor can be craftable with specific recipes, as they're adaptations of existing tools. As they are alternatives to things you could carry anyway, I don't see why you need much gatekeeping.
* Advanced ones - because they're too sci-fi OR because they are too good (night-vision without googles) - would need trade or quest. Maybe a "bring X Y Z and we build for you" interaction.

**Comment by I-am-Erk** [comment]
At this point I will close this one. Some of the points are still relevant but it's OK to just have it in the archive for now: the landscape has changed a lot.
